pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -301,6 +298,7 @@ func (s *FlinkStateMachine) handleApplicationSavepointing(ctx context.Context, a
 		s.flinkController.LogEvent(ctx, application, corev1.EventTypeWarning, "SavepointFailed",
 			fmt.Sprintf("Could not savepoint existing job: %s", reason))
 		application.Status.RetryCount = 0
+		application.Status.JobStatus.JobID = ""
```

Comment:
Any reason to clear the job id, when savepointing is not succeeding?

Reply:
Whoops, that change should not have been left in. Good catch!